### DISCUSSION TITLE
42501 : permission denied for table http_request_queue

### DISCUSSION THREAD

**Discussion by monicakh** [supabase/supabase#21450] [Troubleshooting]
_This is a copy of a troubleshooting article on Supabase's docs site. It may be missing some details from the original. View the [original article](https://supabase.com/docs/guides/troubleshooting/42501--permission-denied-for-table-httprequestqueue-KnozmQ)._

If you're currently blocked by the above error, run the following in your Supabase SQL editor:

*   Check `select * from net.http_request_queue` and make sure it's empty.
*   Try `drop extension pg_net; create extension pg_net schema extensions;`
*   If that doesn't work (e.g. because some objects depend on it), then [contact support](https://supabase.com/support).

**Comment by rigor789** [comment]
These resulted in
```
must be owner of function net.http_get
...
```
However uninstalling/reinstalling `pg_net` seems to have worked as suggested in a Discord thread.

Is the cause known, what can we do to prevent this from happening in the future?

**Reply by vickkhera** [reply]
Reinstalling pg_net does work, but is a wholly unacceptable thing to do on a production instance.  I asked support to run the above SQL for me.

For me the cause was upgrading my instance. The underlying error in the upgrade process is something I'd like to understand too. It seems curious that it could happen at all that a function goes from security definer to invoker.

**Reply by monicakh** [reply]
Unfortunately, as you pointed out the commands provided earlier still require elevated privileges. For now, I've updated the guide with the latest solution.

**Comment by vickkhera** [comment]
The error I received was `permission denied for sequence http_request_queue_id_seq` which didn't find this post when searching, so I'm adding it here in case someone else has it.

**Comment by pcnc** [comment]
Hello everybody, 

We have been actively investigating this issue and have determined that it's an issue stemming from a permission reset triggered as part of the execution of pg_upgrade, the underlying process used to run database upgrades.

We have merged [a fix](https://github.com/supabase/postgres/pull/905) to address this, which has recently been deployed to our production platform.
